### ISSUE TITLE
Cannot compile OpenCV on Mac OS Sierra (Kinetic)

### ISSUE THREAD

**Issue by kvalev** [ros-gbp/opencv3-release#8]
Hi,

when compiling OpenCV on Mac OS Sierra for the Kinetic release I get the following error message:
```
[ 53%] Building CXX object modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_qtkit.mm.o
.../ros_catkin_ws/src/opencv3/modules/videoio/src/cap_qtkit.mm:46:9: fatal error: 'QTKit/QTKit.h' file not found
#import <QTKit/QTKit.h>
        ^
1 error generated.
make[2]: *** [modules/videoio/CMakeFiles/opencv_videoio.dir/src/cap_qtkit.mm.o] Error 1
make[1]: *** [modules/videoio/CMakeFiles/opencv_videoio.dir/all] Error 2
make: *** [all] Error 2
```

I compared the OpenCV source code that was checked out in ros_catkin_ws/src with the one from OpenCV's github and it seems that there are several discrepancies. In my case the detection of QTKit seems to be failing, which was revamped a while ago in OpenCV. Here is the relevant snippet from the `ros_catkin_ws/src/opencv3/cmake/OpenCVFindLibsVideo.cmake` file:

```
# --- QuickTime ---
if (NOT IOS)
  if(WITH_QUICKTIME)
    set(HAVE_QUICKTIME YES)
  elseif(APPLE AND CMAKE_COMPILER_IS_CLANGCXX)
    set(HAVE_QTKIT YES)
  endif()
endif()
```

The same snippet looks **very** differently in [OpenCV 3.2](https://github.com/opencv/opencv/blob/3.2.0/cmake/OpenCVFindLibsVideo.cmake#L275-L286), but is identical to the one in [OpenCV 3.1](https://github.com/opencv/opencv/blob/3.1.0/cmake/OpenCVFindLibsVideo.cmake#L315-L321).

Also going by the commit message of the [upstream/3.2.0 branch](https://github.com/ros-gbp/opencv3-release/tree/upstream/3.2.0) I suspect that you have checked out the wrong version for the 3.2 release.

**Comment by wjwwood** [comment]
@vrabaud fyi

**Comment by vrabaud** [comment]
@kvalev , thx for the detailed explanation. The version shipped as 3.2.0 was indeed wrong. It now seems ok:
https://github.com/ros-gbp/opencv3-release/blob/upstream/3.2.0/cmake/OpenCVFindLibsVideo.cmake#L275
Please confirm.
